webui/react/src/hooks/useSettings.ts

```diff
@@ -230,7 +272,7 @@ const useSettings = <T>(config: SettingsConfig, options?: SettingsHookOptions):
     userSettings,
   } = useStore();
   const prevSearch = usePrevious(location.search, undefined);
-  const [settings, setSettings] = useState<T>(() => getDefaultSettings<T>(config, storage));
+  const [settings, setSettings] = useState<T>(() => getDefaultSettings<T>(config, userSettings));
```

Comment:
Why does this  have a local state for user settings instead of relying on and modifying the global state?